dvc/scm/git.py

```diff
@@ -145,6 +147,116 @@ def branch(self, branch):
     def tag(self, tag):
         self.git.git.tag(tag)
 
+    def open(self, path, branch=None):
+        if not branch:
+            return open(path)
+
+        relpath = os.path.relpath(path, self.root_dir)
+
+        # XXX: make git_object_by_path working with relative paths?
+        obj = self.git_object_by_path(path, branch=branch)
+        if obj is None:
+            msg = "No such file in branch '%s'" % branch
+            raise IOError(2, msg, relpath)
+        if obj.mode == self.GIT_MODE_DIR:
+            raise IOError(21, "Is a directory", relpath)
+        # ensure that it is a file, and file could be 755 also
+        if obj.mode & self.GIT_MODE_FILE != self.GIT_MODE_FILE:
+            raise IOError(21, "Is not a file", relpath)
+
+        # git object data_stream is a fragile thing, it is better to read it
+        # immediately, also it needs to be to decoded if we follow the
+        # `open()` behaviour
+        from io import StringIO
```

Comment:
any specific reason to place this import here?

Reply:
Also, use `from dvc.utils.compat import StringIO`, so that all builds pass